lib/segment/src/segment/mod.rs

```diff
@@ -105,6 +105,19 @@ impl VectorData {
 
         index_task.into_iter().chain(storage_task)
     }
+
+    pub fn size_of_available_vectors_in_bytes(
+        storage: &VectorStorageEnum,
+        index: &VectorIndexEnum,
+    ) -> usize {
+        let size = storage
+            .size_of_available_vectors_in_bytes()
+            .or_else(|| index.size_of_searchable_vectors_in_bytes());
+
+        debug_assert!(size.is_some());
+
+        size.unwrap_or(0)
```

Comment:
If we were to go with this design, we would need a log warning here to know something is wrong

Reply:
Ditched this thang